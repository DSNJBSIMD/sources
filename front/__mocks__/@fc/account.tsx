import type { PropsWithChildren } from 'react';
import React from 'react';

export const AccountContext = React.createContext(expect.any(Object));

export const AccountProvider = jest.fn(
  ({ children, config }: { config: unknown } & PropsWithChildren) => (
    <AccountContext.Provider value={config}>
      <div data-mockid="AccountProvider">
        <div>{children}</div>
      </div>
    </AccountContext.Provider>
  ),
);

export const ConnectValidator = {
  validate: jest.fn(),
};

export const Options = {
  CONFIG_NAME: 'Account',
};
