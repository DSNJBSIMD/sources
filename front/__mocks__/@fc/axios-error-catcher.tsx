import type { PropsWithChildren } from 'react';
import React from 'react';

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));

export const AxiosErrorCatcherProvider = jest.fn(
  ({ children, value }: { value: unknown } & PropsWithChildren) => (
    <AxiosErrorCatcherContext.Provider value={value}>{children}</AxiosErrorCatcherContext.Provider>
  ),
);
