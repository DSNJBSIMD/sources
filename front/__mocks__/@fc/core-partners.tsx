export const CreateInstanceButton = jest.fn(() => <div data-mockid="CreateInstanceButton" />);

export const InstanceComponent = jest.fn(() => <div data-mockid="InstanceComponent" />);

export const InstancesListComponent = jest.fn(() => <div data-mockid="InstancesListComponent" />);

export const PartnersService = {
  commit: jest.fn(),
  get: jest.fn(),
};

export enum Environment {
  SANDBOX = 'SANDBOX',
  PRODUCTION = 'PRODUCTION',
}

export enum Options {
  CONFIG_NAME = 'Partners',
}

export enum PublicationStatus {
  DRAFT = 'DRAFT',
  PENDING = 'PENDING',
  PUBLISHED = 'PUBLISHED',
}
