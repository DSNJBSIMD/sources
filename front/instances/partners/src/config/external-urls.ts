export const ExternalUrls: Record<string, string> = {
  spConfigurationDocUrl:
    'https://docs.partenaires.franceconnect.gouv.fr/fs/devenir-fs/projet-bac-a-sable/#configuration-de-votre-fournisseur-de-service',
};
