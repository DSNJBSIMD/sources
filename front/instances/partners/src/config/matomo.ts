import type { MatomoConfig } from '@fc/matomo';

export const Matomo: MatomoConfig = {
  siteId: 309,
  url: 'https://stats.data.gouv.fr/',
};
