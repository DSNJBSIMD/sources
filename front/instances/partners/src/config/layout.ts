import { LogoFranceConnect } from '@fc/assets';
import type { LayoutConfig } from '@fc/layout';

export const Layout: LayoutConfig = {
  features: {
    showLicence: false,
    showServiceTitle: true,
  },
  footer: {
    description:
      'L’Espace partenaires FranceConnect est un dispositif qui permet aux partenaires de gérer leur fournisseur de service FranceConnect.',
    links: [
      {
        external: true,
        href: 'https://www.legifrance.gouv.fr',
        label: 'legifrance.gouv.fr',
        title: 'Accéder au site legifrance.gouv.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://www.info.gouv.fr',
        label: 'info.gouv.fr',
        title: 'Accéder au site info.gouv.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://www.service-public.fr/',
        label: 'service-public.fr',
        title: 'Accéder au site service-public.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://data.gouv.fr',
        label: 'data.gouv.fr',
        title: 'Accéder au site data.gouv.fr nouvelle fenêtre',
      },
    ],
    navigation: [
      {
        external: false,
        href: '/plan-du-site',
        label: 'Plan du site',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/accessibilite',
        label: 'Accessibilité : non conforme',
      },
      {
        external: false,
        href: '/mentions-legales',
        label: 'Mentions légales',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Données personnelles',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Gestion des cookies',
      },
    ],
  },
  service: {
    baseline: 'Espace Partenaires',
    homepage: '/',
    logo: LogoFranceConnect,
    name: 'FranceConnect',
  },
  sitemap: [
    {
      href: '/',
      label: 'Accueil',
    },
    {
      href: '/mentions-legales',
      label: 'Mentions Légales',
    },
    {
      external: true,
      href: 'https://franceconnect.gouv.fr/licence',
      label: 'Licence',
    },
    {
      external: true,
      href: 'https://franceconnect.gouv.fr/accessibilite',
      label: 'Accessibilité',
    },
  ],
};
