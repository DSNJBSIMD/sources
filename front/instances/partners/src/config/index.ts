import { Account } from './account';
import { DTO2Form } from './dto2form';
import { ExternalUrls } from './external-urls';
import { HttpClient } from './http-client';
import { Layout } from './layout';
import { Matomo } from './matomo';
import { Partners } from './partners';

export const AppConfig = {
  Account,
  DTO2Form,
  ExternalUrls,
  HttpClient,
  Layout,
  Matomo,
  Partners,
};
