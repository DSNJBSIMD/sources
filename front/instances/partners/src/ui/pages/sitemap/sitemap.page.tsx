import classnames from 'classnames';
import React from 'react';

import { ConfigService } from '@fc/config';
import { LinkComponent } from '@fc/dsfr';
import type { LayoutConfig } from '@fc/layout';
import { Options } from '@fc/layout';

import classes from './sitemap.module.scss';

export const SitemapPage = React.memo(() => {
  const { sitemap = [] } = ConfigService.get<LayoutConfig>(Options.CONFIG_NAME);

  return (
    <nav className="fr-col-12 fr-col-md-8" id="plan-du-site">
      <h1>Plan du site</h1>
      <ul className="fr-raw-list">
        {sitemap.map((link) => {
          const { external, href, label, title } = link;
          return (
            <li key={label} className={classnames(classes.blueLink, 'fr-mb-2w')}>
              <LinkComponent external={external} href={href} title={title}>
                {label}
              </LinkComponent>
            </li>
          );
        })}
      </ul>
    </nav>
  );
});

SitemapPage.displayName = 'SitemapPage';
