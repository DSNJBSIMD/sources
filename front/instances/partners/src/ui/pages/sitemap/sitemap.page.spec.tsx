import { render } from '@testing-library/react';

import type { NavigationLinkInterface } from '@fc/common';
import { ConfigService } from '@fc/config';
import { LinkComponent } from '@fc/dsfr';
import type { LayoutConfig } from '@fc/layout';

import { SitemapPage } from './sitemap.page';

describe('SitemapPage', () => {
  beforeEach(() => {
    // Given
    jest.mocked(ConfigService.get).mockReturnValue({} as unknown as LayoutConfig);
  });

  it('should match the snapshot, when sitemap prop is not defined', () => {
    // When
    const { container } = render(<SitemapPage />);

    // Then
    expect(container).toMatchSnapshot();
    expect(LinkComponent).not.toHaveBeenCalled();
  });

  it('should match the snapshot, when sitemap prop into layout config is defined', () => {
    // Given
    const sitemapMock = [
      {
        external: false,
        href: '/any-href-mock-1',
        label: 'any-label-mock-1',
        title: 'any-title-mock-1',
      },
      {
        external: true,
        href: '/any-href-mock-2',
        label: 'any-label-mock-2',
        title: 'any-title-mock-2',
      },
    ] as NavigationLinkInterface[];
    jest.mocked(ConfigService.get).mockReturnValueOnce({ sitemap: sitemapMock } as LayoutConfig);

    // When
    const { container } = render(<SitemapPage />);

    // Then
    expect(container).toMatchSnapshot();
    expect(LinkComponent).toHaveBeenCalledTimes(2);
    expect(LinkComponent).toHaveBeenNthCalledWith(
      1,
      {
        children: 'any-label-mock-1',
        external: false,
        href: '/any-href-mock-1',
        title: 'any-title-mock-1',
      },
      undefined,
    );
    expect(LinkComponent).toHaveBeenNthCalledWith(
      2,
      {
        children: 'any-label-mock-2',
        external: true,
        href: '/any-href-mock-2',
        title: 'any-title-mock-2',
      },
      undefined,
    );
  });
});
