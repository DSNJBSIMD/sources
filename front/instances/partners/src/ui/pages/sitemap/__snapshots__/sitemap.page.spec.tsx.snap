// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SitemapPage should match the snapshot, when sitemap prop into layout config is defined 1`] = `
<div>
  <nav
    class="fr-col-12 fr-col-md-8"
    id="plan-du-site"
  >
    <h1>
      Plan du site
    </h1>
    <ul
      class="fr-raw-list"
    >
      <li
        class="blueLink fr-mb-2w"
      >
        <span
          data-mockid="LinkComponent"
        />
      </li>
      <li
        class="blueLink fr-mb-2w"
      >
        <span
          data-mockid="LinkComponent"
        />
      </li>
    </ul>
  </nav>
</div>
`;

exports[`SitemapPage should match the snapshot, when sitemap prop is not defined 1`] = `
<div>
  <nav
    class="fr-col-12 fr-col-md-8"
    id="plan-du-site"
  >
    <h1>
      Plan du site
    </h1>
    <ul
      class="fr-raw-list"
    />
  </nav>
</div>
`;
